addons/interaction/functions/fnc_switchAnimActions.sqf

```diff
@@ -81,61 +81,60 @@ private _condition = {
 
     GVAR(animActionsEnabled)
     && {isClass _animConfig}
-    && {0 != [_animConfig >> "scope", "number", 1] call CBA_fnc_getConfigEntry}
-    && {0 != [_vehicleConfig >> QGVAR(anims) >> _anim >> "enabled", "number", 1] call CBA_fnc_getConfigEntry}
+    && {0 != [_animConfig >> "scope", "number", 1] call CBA_fnc_getConfigEntry} // anim hidden
+    && {isNumber (_animConfig >> "initPhase")} // anim correct - CUP inherits and clears some
+    && {0 != [_vehicleConfig >> QGVAR(anims) >> _anim >> "enabled", "number", 1] call CBA_fnc_getConfigEntry} // anim disabled
     && {_target animationPhase _anim != _phase}
     && {[_player, _target, ["isNotSwimming"]] call EFUNC(common,canInteractWith)}
 };
 
 {
-    if (isClass (_x >> QGVAR(anims))) then {
-        private _vehicleConfig = _x;
-        private _vehicle = configName _vehicleConfig;
-        {
-            scopeName "animLoop";
-            private _anim = configName _x;
-            private _animConfigParent = inheritsFrom _vehicleConfig >> QGVAR(anims) >> _anim;
-            if (
-                !isClass _animConfigParent
-                || {_x != _animConfigParent && {0 == [_animConfigParent >> "inherit", "number", 1] call CBA_fnc_getConfigEntry}}
-            ) then {
-                private _positions = [_x >> "position"] call BIS_fnc_getCfgData;
-                if (!isNil "_positions") then {
-                    if (_positions isEqualType "") then {
-                        _positions = [compile format [
-                            QUOTE(call compile getText (configFile >> 'CfgVehicles' >> typeOf _target >> QQGVAR(anims) >> '%1' >> 'position')),
-                            _anim
-                        ]];
-                    };
-                } else {
-                    if ("" == getText (_x >> "selection")) then {
-                        ERROR_2("No action position for vehicle %1 anim %2",_vehicle,_anim);
-                        breakTo "animLoop";
-                    };
+    private _vehicleConfig = _x;
+    private _vehicle = configName _vehicleConfig;
+    {
+        scopeName "animLoop";
+        private _anim = configName _x;
+        private _animConfigParent = inheritsFrom _vehicleConfig >> QGVAR(anims) >> _anim;
+        if (
+            !isClass _animConfigParent
+            || {_x != _animConfigParent && {0 == [_animConfigParent >> "inherit", "number", 1] call CBA_fnc_getConfigEntry}}
+        ) then {
+            private _positions = [_x >> "position"] call BIS_fnc_getCfgData;
+            if (!isNil "_positions") then {
+                if (_positions isEqualType "") then {
                     _positions = [compile format [
-                        QUOTE(_target selectionPosition getText (configFile >> 'CfgVehicles' >> typeOf _target >> QQGVAR(anims) >> '%1' >> 'selection')),
+                        QUOTE(call compile getText (configFile >> 'CfgVehicles' >> typeOf _target >> QQGVAR(anims) >> '%1' >> 'position')),
```

Comment:
Is this really needed? there is a compile in L105. Can't we just do
call {+text+} instead?

Reply:
Also why call compile here but not in L116?

Reply:
L105 is for `position = "[1,1.4,0]"` case. It produces this position code:
`{call compile getText (configFile >> 'CfgVehicles' >> typeOf _target >> "ace_interaction_anims" >> 'hide_spare' >> 'position')}`
so any child class with different `position` value will use exactly that value and not from parent class.

L116 is for `selection = "spare"` case. It produces this position code:
`{_target selectionPosition getText (configFile >> 'CfgVehicles' >> typeOf _target >> "ace_interaction_anims" >> 'spare' >> 'selection')}`
so child class with different selection position will use its own value and not position from parent class.

Reply:
>L105 is for position = "[1,1.4,0]" case.

parseSimpleArray.
`call compile` is almost always a mistake and just bad code. Only valid reason would be if position is in-config code string that's intended to be evaluated.

Reply:
it was implemented this way to reflect `position` in [ace interaction config ](https://ace3mod.com/wiki/framework/interactionMenu-framework.html#2-adding-actions-via-config) and is really intended to be evaluated. `parseSimpleArray` could be used here but it will restrict only `[1,2,3]`-like values.

Reply:
Yeah, interaction menu can take code to evaluate, I think this is fine.